FreestarAds-Ogury/Source/objc/OguryThumbnailMediator.m

```diff
@@ -0,0 +1,141 @@
+//
+//  OguryThumbnailMediator.m
+//  FreestarAds-Ogury
+//
+//  Created by Carlos Alcala on 8/08/22.
+//
+
+#import "OguryThumbnailMediator.h"
+#import <UIKit/UIKit.h>
+#import <OguryChoiceManager/OguryChoiceManager.h>
+#import <OguryAds/OguryAds.h>
+
+@interface OguryThumbnailMediator()<OguryAdsThumbnailAdDelegate, FSTRMediatorEnabling>
+
+@property (nonatomic, strong) OguryAdsThumbnailAd *thumbnail;
+@property CGSize requestedSize;
+@property CGPoint startPoint;
+@property(nonatomic, assign) FreestarThumbnailAdSize freestarAdSize;
+
+@end
+
+@implementation OguryThumbnailMediator
+
+- (BOOL)isEnabledForMediatorFormat:(FSTRMediatorFormatType)type {
+    switch (type) {
+        case FSTRMediatorFormatTypeThumbnail:
+            return YES;
+        default:
+            return NO;
+    }
+}
+
+- (BOOL)canShowThumbnailAd {
+    return YES;
+}
+
+-(void)loadThumbnailAd {
+    FSTRLog(@"OGURY: loadThumbnailAd");
+    FSTRLog(@"OGURY: adunitId %@", [self.mPartner adunitId]);
+
+    if ([self.mPartner adunitId] == nil) {
+        [self partnerAdLoadFailed:@"adunitId is nil"];
+        return;
+    }
+
+    self.startPoint = CGPointMake(100, 500);
+    self.thumbnail = [[OguryAdsThumbnailAd alloc]initWithAdUnitID:[self.mPartner adunitId]];
```

Comment:
@carlosalcala-freestar  do u get deprecating warnings?  if so, please use the latest ogury APIs... thx.

Reply:
Fixing those @maddapper